World.py

```diff
@@ -100,6 +95,15 @@ def __init__(self, id, settings):
 
         self.resolve_random_settings()
 
+        if self.open_forest == 'closed' and self.starting_age == 'adult' and self.logic_rules == 'glitchless' and \
+            not (self.open_door_of_time or ('Ocarina' and 'Song of Time') in self.distribution.starting_items or \
+            self.shuffle_special_interior_entrances or self.shuffle_overworld_entrances or self.spawn_positions):
+                # adult is not compatible with glitchless closed forest without shuffled entrances or open door of time
```

Comment:
A comment explaining why this is the case would be nice. It took us a while to figure out why the Door of Time setting affects this #dev-glitchless-logic.